src/render/batch.rs

```diff
@@ -252,7 +254,7 @@ pub struct CoreBatch<T: ShaderParam> {
     state_id: Id<DrawState>,
 }
 
-impl<T: ShaderParam> Copy for CoreBatch<T> where T::Link: Copy {}
+impl<T: Clone + ShaderParam> Copy for CoreBatch<T> where T::Link: Copy {}
```

Comment:
I assume the need for the `Clone` bound would go away after my previous comment.